### PR TITLE
[ENH] Validate filter tags in list_models and list_datasets

### LINKED ISSUE #2920: [ENH] Throw error for incorrect argument in `list_models` and `list_datasets` methods
The `list_models` and `list_datasets` methods provide filtering options using the available tags in their specific classes.  For example:
```python
In [12]: from pgmpy.example_models import list_models

In [13]: list_models(n_nodes=10)
Out[13]: 
['bnrep/APSsystem',
 'bnrep/argument',
 'bnrep/gonorrhoeae',
 'bnrep/greencredit',
 'bnrep/lawschool',
 'bnrep/liquefaction',
 'bnrep/liquidity',
 'bnrep/nuclearwaste',
 'bnrep/salmonella2',
 'bnrep/seismic',
 'bnrep/tbm']
```
However, if an incorrect tag is provided, the method fails silently and returns no models/datasets. For example:
```python
In [14]: list_models(n_nodes=46, is_paraterized=True)
Out[14]: []
```
In the example above, there is a typo in the tag `is_paraterized`. Ideally, our code should check if the passed argument is one of the available tags. And if not it should throw a meaningful error to let the user know that the argument is not recognized.

### PR DESCRIPTION
**Pull request by kajal-jotwani**
# The following checklist is mandatory.

Your PR will be closed if you remove the checklist or do not answer the questions to a satisfactory level. Use of LLM is **strictly forbidden** for any part of this checklist (even for improving language).

### Your checklist for this pull request

- [x] Have you followed all the steps from our [Contributing Guide](https://github.com/pgmpy/pgmpy/blob/dev/Contributing.md)?
- [x] Does the PR fully address the linked issue and is within its defined scope? If you are still working on the PR, mark it as draft.
- [x] Are all the GitHub Actions checks passing? If not, mark your PR as draft while you fix it.

Please answer the following questions:

- Did you use an LLM for any assistance? Please describe how and what you used it for?
Yes, I used an LLM to help check my approach and assist with writing some of the tests.
- What steps have you taken to verify that the changes correctly address the issue? And what edge cases have you considered?
I added tests and also checked the changes in the Python REPL to confirm that invalid tags raise an error and valid filters still work.
- Has the LLM added try-except blocks? They will need to be removed; any error handling must be explicit.
No try-except blocks are added
- Have you used LLM for generating tests? They need to be compressed into a smaller number of tests without reducing coverage.
yes and there are only small tests added 

### Issue number(s) that this pull request fixes
- Fixes #2920 

### List of changes to the codebase in this pull request
Previously, passing an incorrect filter tag would silently return an empty list. This PR adds validation for filter tags in `list_models` and `list_datasets` and raises a `ValueError` when unrecognized tags are provided. The tests are also added for the behaviour

```
(venv) kajal@kajal-HP-Pavilion-Laptop-15-eh2xxx:~/pgmpy$ python
Python 3.12.3 (main, Jan 22 2026, 20:57:42) [GCC 13.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
Ctrl click to launch VS Code Native REPL
>>> from pgmpy.example_models import list_models
>>> from pgmpy.datasets import list_datasets
>>> list_models(n_nodes=10)
['bnrep/APSsystem', 'bnrep/argument', 'bnrep/gonorrhoeae', 'bnrep/greencredit', 'bnrep/lawschool', 'bnrep/liquefaction', 'bnrep/liquidity', 'bnrep/nuclearwaste', 'bnrep/salmonella2', 'bnrep/seismic', 'bnrep/tbm']
>>> list_models(is_paraterized=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kajal/pgmpy/pgmpy/example_models/_base.py", line 215, in list_models
    raise ValueError(
ValueError: Unrecognized filter argument(s): ['is_paraterized']. Valid filter tags are: ['is_continuous', 'is_discrete', 'is_hybrid', 'is_parameterized', 'n_edges', 'n_nodes', 'name'].
>>> list_models(num_nodes=10)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kajal/pgmpy/pgmpy/example_models/_base.py", line 215, in list_models
    raise ValueError(
ValueError: Unrecognized filter argument(s): ['num_nodes']. Valid filter tags are: ['is_continuous', 'is_discrete', 'is_hybrid', 'is_parameterized', 'n_edges', 'n_nodes', 'name'].
>>> list_models(is_paraterized=True, n_nodez=10, unknown_arg="test")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kajal/pgmpy/pgmpy/example_models/_base.py", line 215, in list_models
    raise ValueError(
ValueError: Unrecognized filter argument(s): ['is_paraterized', 'n_nodez', 'unknown_arg']. Valid filter tags are: ['is_continuous', 'is_discrete', 'is_hybrid', 'is_parameterized', 'n_edges', 'n_nodes', 'name'].
>>> 
```

### CODE REVIEW

**Comment by ankurankan** [comment]
@kajal-jotwani 
1. Please do not use screenshots anywhere and instead copy-paste code. We can not copy text from a screenshot.
2. You have not filled in the PR checklist

File: pgmpy/tests/test_datasets/test_datasets.py [hunk 1]
```diff
@@ -109,3 +109,14 @@ def test_load_covariance_dataset():
...
+
+def test_invalid_tag():
+    with pytest.raises(ValueError, match="Unrecognized filter argument"):
+        list_datasets(is_paraterized=True)  # typo
+
+    with pytest.raises(ValueError, match="Unrecognized filter argument"):
+        list_datasets(num_samples=100)  # wrong key name entirely
+
+    result = list_datasets(is_discrete=True)
+    assert result
```
Comment (ankurankan) [inline pgmpy/tests/test_datasets/test_datasets.py]: Isn't this a repeated test? The filtering has been tested somewhere else as well?

File: pgmpy/tests/test_example_models/test_example_models.py [hunk 2]
```diff
@@ -309,6 +309,17 @@ def test_list_models():
...
 
+def test_invalid_tag():
+    with pytest.raises(ValueError, match="Unrecognized filter argument"):
+        list_models(is_paraterized=True)  # typo
+
+    with pytest.raises(ValueError, match="Unrecognized filter argument"):
+        list_models(num_nodes=10)  # wrong key name entirely
+
+    result = list_models(is_discrete=True)
+    assert result
```
Comment (ankurankan) [inline pgmpy/tests/test_example_models/test_example_models.py]: Same

File: pgmpy/datasets/_base.py [hunk 3]
```diff
@@ -314,6 +314,15 @@ def list_datasets(**filter_tags) -> list[str]:
     >>> list_datasets(is_discrete=True, has_ground_truth=True)
     ['sachs_discrete']
     """
+    valid_tags = set(_BaseDataset._tags.keys())
+    invalid_tags = set(filter_tags.keys()) - valid_tags
+
+    if invalid_tags:
```
Comment (ankurankan) [inline pgmpy/datasets/_base.py]: Might be cleaner to use the walrus operator.

File: pgmpy/example_models/_base.py [hunk 4]
```diff
@@ -208,6 +208,15 @@ def list_models(**filter_tags) -> list[str]:
     >>> list_models(is_parameterized=False)
     ['dagitty/acid_1996', ...., ]
     """
+    valid_tags = set(_BaseExampleModel._tags.keys())
+    invalid_tags = set(filter_tags.keys()) - valid_tags
+
+    if invalid_tags:
```
Comment (ankurankan) [inline pgmpy/example_models/_base.py]: Same here.

**Review by ankurankan — CHANGES_REQUESTED** [review]

File: pgmpy/tests/test_datasets/test_datasets.py (hunk 1 above)
Comment (kajal-jotwani) [reply pgmpy/tests/test_datasets/test_datasets.py]: yes its there in `test_list_datasets`  I've removed it

File: pgmpy/datasets/_base.py (hunk 3 above)
Comment (kajal-jotwani) [reply pgmpy/datasets/_base.py]: I've updated it to use the walrus operator now, thanks!

**Review by ankurankan — APPROVED** [review]
@kajal-jotwani Thanks for the PR. Looks good.
